web/inc/main.php

```diff
@@ -1,6 +1,12 @@
 <?php
 session_start();
 
+if ($_SERVER["HTTP_EARLY_DATA"] === "1") {
+	# Protection against early_data / 0-rrt
+	http_response_code(425);
+	exit();
+}
+
```

Comment:
@jaapmarcus No need for this.

Seems you haven't fully understand how it works, don't hit me...:zany_face:

Potentially risky requests with early data never reach the backend.

Reply:
I'll make some changes later to comply with RFC 8470, to avoid access failures caused by such application processing logic.

Reply:
I just did some testing:
https://internet.nl
And https://nabla-c0d3.github.io/sslyze/documentation/ and both still seem to report early_data enabled. Probably internet.nl uses sslyze as background. And sslyze uses an invalid request method that reports 400.

Also tested it with 
```
host=hestiacp.com # replace with your server name 
echo -e "HEAD / HTTP/1.1\r\nHost: $host\r\nConnection: close\r\n\r\n" > request.txt
openssl s_client -connect $host:443 -tls1_3 -sess_out session.pem -ign_eof < request.txt
openssl s_client -connect $host:443 -tls1_3 -sess_in session.pem -early_data request.txt
```
And the same issue but adding the php code solves the issue..

(Replaced hestiacp.com with my own domain)

Reply:
If an attacker copies the 0-RTT request data (encrypted, with request method non-idempotent or GET with query)

Then attacker makes a replay attempt, but no matter how many times its replayed (when PSK is not expired)

NGINX will always return 30X/425 (encrypted)

No data will reach the backend.